zutil.c

```diff
@@ -109,3 +109,52 @@ void Z_INTERNAL zng_cfree(void *opaque, void *ptr) {
     Z_UNUSED(opaque);
     zng_free(ptr);
 }
+
+/* Since we support custom memory allocators, some which might not align memory as we expect,
+ * we have to ask for extra memory and return an aligned pointer. */
+void Z_INTERNAL *zng_calloc_aligned(zng_calloc_cb calloc, void *opaque, unsigned items, unsigned size, unsigned align) {
+    uintptr_t return_ptr, original_ptr;
+    int32_t alloc_size, align_diff;
+    void *ptr;
+
+    /* Allocate enough memory for proper alignment and to store the original memory pointer */
+    alloc_size = sizeof(void *) + (items * size) + align;
+
+    if (calloc) {
```

Comment:
_might_ want to make sure this is resolving the pointer and not the calloc function.  Assuming you don't include stdlib.h anywhere on the include path this _shouldn't_ be the case, but it's possible.  Or maybe precedence makes the parameter name shadow the actual libc function.  In any case, it might behoove us to rename that parameter.

Reply:
I have renamed it, but I have also removed the if statement, since even before zalloc is called, it is defaulted to zng_alloc, etc.

Reply:
This then presents another issue though, no?  If we called zng_alloc, we definitely don't need to over allocate, as posix_memalign is handling this for us.

Reply:
`zalloc` and `zfree` are never NULL because we set it to default value of `zng_calloc` and `zng_cfree` in `deflateInit` and `inflateInit`.

I have added a check that will just use `zng_calloc/cfree` if not using custom allocators.